staging/src/k8s.io/apimachinery/pkg/api/validate/each.go

```diff
@@ -33,12 +34,12 @@ type CompareFunc[T any] func(T, T) bool
 func EachSliceVal[T any](ctx context.Context, op operation.Operation, fldPath *field.Path, newSlice, oldSlice []T,
 	cmp CompareFunc[T], validator ValidateFunc[*T]) field.ErrorList {
 	var errs field.ErrorList
-	for i, val := range newSlice {
+	for _, val := range newSlice {
 		var old *T
 		if cmp != nil && len(oldSlice) > 0 {
 			old = lookup(oldSlice, val, cmp)
 		}
-		errs = append(errs, validator(ctx, op, fldPath.Index(i), &val, old)...)
+		errs = append(errs, validator(ctx, op, fldPath.Key(fmt.Sprintf("name=%v", val)), &val, old)...)
```

Comment:
In the hand-written validation we have today, we always (or almost always?) use indices in field paths (for both plain slices and for `listType=map`).  Changing that as part of switching to declarative validation will make migrating field more difficult since we need to correlated the hand-written validation errors with declarative validation errors, and one of components of an error we check is the field path.

If we ever did switch to logical keys, we'd need to only include the `listMapKey` fields, and we'd need to continue to use the index for plain slices.

cc @aaron-prindle @thockin for thoughts on direction

Reply:
I think this is the most important aspect.  I think this will break the error-correlation logic, and will require us to "fix" hand-written code at the same time, which is something we are striving to NOT do (compartmentalize the risk).

Most places in our API do not have huge lists, so this is (IMO) not super important.  Is there some real use-case this is solving for?